widgetry/src/widgets/button.rs

```diff
@@ -433,6 +442,15 @@ impl<'b, 'a: 'b, 'c> ButtonBuilder<'a, 'c> {
         self
     }
 
+    /// Set a tooltip [`Text`] to appear when hovering over the button, when the button is
+    /// disabled.
+    ///
+    /// This tooltip is only displayed when `disabled(true)` is also called.
+    pub fn disabled_tooltip(mut self, tooltip: impl Into<Text>) -> Self {
```

Comment:
Not sure this API is best. I do think disabled buttons warrant a separate tooltip. I thought about making `disabled_tooltip` imply `disabled(true)` or having the API something like `.disabled(Option<Text>)`, but all of those feel more confusing

Reply:
My first impulse was to have an api like: 
```
btn_builder
    .tooltip("normal tooltip", ControlState::Default)
    .tooltip("tooltip while disabled", ControlState::Disabled)
    .disabled(app.primary.map.get_r_edit(r) == orig_road_state)
```

backed by an implementation like:
```
fn tooltip(mut self, tooltip: impl Into<Text>, forState: ControlState) {
    self.style_mut(forState).tooltip = Some(tooltip.into());
    self
}
```

But that'd be a little weird for a couple reasons... would we treat `ControlState::Default` vs. `ControlState::Hovered` the same? Or would we not use `ControlState::Default` for tooltips?

A lesser concern would be that it'd be more verbose for the default case.


So anyway, I think where you landed makes more sense.

Reply:
> would we treat ControlState::Default vs. ControlState::Hovered the same? Or would we not use ControlState::Default for tooltips?

I didn't even think of it in terms of `ControlState`, but yeah, that would be confusing! Setting it for `ControlState::Default` means if the button exists, no matter where the cursor is, it's haunted by a tooltip! ;)